pkg/R/DataFrame.R

```diff
@@ -253,6 +253,96 @@ setMethod("unpersist",
             x
           })
 
+#' Repartition
+#'
+#' Return a new DataFrame that has exactly numPartitions partitions.
+#'
+#' @param x A SparkSQL DataFrame
+#' @param numPartitions The number of partitions to use.
+#' @rdname repartition
+#' @export
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' sqlCtx <- sparkRSQL.init(sc)
+#' path <- "path/to/file.json"
+#' df <- jsonFile(sqlCtx, path)
+#' newDF <- repartition(df, 2L)
+#'}
+
+setGeneric("repartition", function(x, numPartitions) { standardGeneric("repartition") })
+
+#' @rdname repartition
+#' @export
+setMethod("repartition",
+          signature(x = "DataFrame", numPartitions = "integer"),
```

Comment:
Actually, I think this will work:
```R
numToInt <- function(num) {
  if (!is.integer(num)) {
    warning(paste("Coercing", as.list(sys.call())[[2]], "to integer."))
    as.integer(num)
  }
}
```